tests/unittests/sources/test_azure.py

```diff
@@ -4487,6 +4654,37 @@ def test_imds_failure_results_in_provisioning_failure(self):
         assert len(self.mock_kvp_report_success_to_host.mock_calls) == 0
 
 
+class TestCheckAzureProxyAgent:
+    @pytest.fixture(autouse=True)
+    def proxy_setup(self, azure_ds, mock_subp_subp):
+        self.azure_ds = azure_ds
+        self.mock_subp_subp = mock_subp_subp
+
```

Comment:
consider adding a mock for the failure report and verify the reports are as expected

Reply:
done